moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -93,6 +93,21 @@ void RobotTrajectory::append(const RobotTrajectory& source, double dt)
     duration_from_previous_[index] += dt;
 }
 
+void RobotTrajectory::append(const RobotTrajectory& source, double dt, size_t start_index, size_t end_index)
```

Comment:
Can "end_index" be guessed for convenience? `end_index = 0` leads to `end_index = source.waypoints_.size()`

Reply:
I am not sure, if I get your point. Are you talking about default arguments? start_index=0 would make sense, but end_index=0 should lead to an empty append trajectory in my opinion. I think that is more intuitive.

Reply:
What about `end_index = std::numeric_limits<std::size_t>::max()`?

Reply:
As soon as we have default args, for both `start_index` and `end_index`, you can drop the old `append()` method completely.

Reply:
Sounds good.

Reply:
> What about `end_index = std::numeric_limits<std::size_t>::max()`?

0 would have been just fine here as a special case.
It does not make sense to ask to copy no element at all.
But the `max` solution is fine to.